src/devtools/client/debugger/src/components/TestInfo/TestInfo.tsx

```diff
@@ -0,0 +1,82 @@
+import { SourceLocation } from "graphql";
+import MaterialIcon from "ui/components/shared/MaterialIcon";
+import { useFetchCypressSpec } from "ui/hooks/useFetchCypressSpec";
+import { getRecordingDuration } from "ui/reducers/timeline";
+import { useAppDispatch, useAppSelector } from "ui/setup/hooks";
+import { TestItem, TestResult } from "ui/types";
+import { getFormattedTime } from "ui/utils/timeline";
+import { selectLocation } from "../../actions/sources";
+import { getThreadContext } from "../../selectors";
+
+export default function TestInfo({
+  spec,
+  testCases,
+  result,
+}: {
+  testCases: TestItem[];
+  spec?: string;
+  result?: TestResult;
+}) {
+  const recordingDuration = useAppSelector(getRecordingDuration);
+  const cypressResults = useFetchCypressSpec();
+
+  return (
+    <div className="flex flex-col px-4 py-2 space-y-1">
+      <div className="flex justify-between">
+        <div className="flex items-center gap-1">
+          {result ? <Status result={result} /> : null}
+          <div className="font-bold">{spec}</div>
+        </div>
+        {recordingDuration ? <div>{getFormattedTime(recordingDuration)}</div> : null}
+      </div>
+      <div className="flex flex-col px-4 py-2 space-y-1">
+        {testCases.map((t, i) => (
+          <TestCase test={t} key={i} location={cypressResults?.[i].location} />
+        ))}
+      </div>
+    </div>
+  );
+}
+
+function TestCase({ test, location }: { test: TestItem; location?: SourceLocation }) {
+  const dispatch = useAppDispatch();
+  const cx = useAppSelector(getThreadContext);
+
+  const onClick = () => {
+    if (location) {
+      dispatch(selectLocation(cx, location as any));
+    }
+  };
+
+  return (
+    <div className="flex flex-col gap-1">
+      <button className="flex flex-row items-center gap-1 group" onClick={onClick}>
+        <Status result={test.result} />
+        <div className="overflow-hidden whitespace-pre overflow-ellipsis group-hover:underline">
+          {test.title}
+        </div>
+      </button>
+      {test.error ? (
+        <div className="font-mono ml-7">
+          <div className="overflow-hidden overflow-ellipsis">{test.error.message}</div>
```

Comment:
I think you want "truncate" for the className. If we do, we should make it toggleable so the user can see the full error message.